mlir/lib/Dialect/Rock/Transforms/GemmOutputSwizzle.cpp

```diff
@@ -0,0 +1,571 @@
+//===- GemmOutputSwizzle - MLIR Rock ops lowering passes -----===//
```

Comment:
naming: Its not just "Gemm".  This should generally work for any kernel that writes to global memory in the end.